tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py

```diff
@@ -0,0 +1,469 @@
+        elif self.verbosity == 2:
+            verbose_eval = 50
+        elif self.verbosity == 3:
+            verbose_eval = 10
+        else:
+            verbose_eval = 1
+
+        for e in range(epochs):
+            if e == 0:
+                logger.log(15, "TabTransformer architecture:")
+                logger.log(15, str(self.model))
+
+            # Whether or not we want to suppress output based on our verbosity.
+            databar_disable = False if e % verbose_eval == 0 else True
+
+            train_loss, val_metric = self._epoch(self.model, trainloader=trainloader, valloader=valloader, y_val=y_val,
+                                                 optimizers=optimizers, loss_criterion=loss_criterion, \
+                                                 pretext=pretext, state=state, scheduler=None, epoch=e, epochs=epochs,
+                                                 databar_disable=databar_disable, params=self.params)
+
+            # Early stopping for pretrain'ing based on loss.
+            if state == 'pretrain':
+                if train_loss < best_loss or e == 0:
+                    if train_loss < best_loss:
+                        best_loss = train_loss
+                    best_val_epoch = e
+            else:
+                if val_metric >= best_val_metric or e == 0:
+                    if valloader is not None:
+                        if not np.isnan(val_metric):
+                            best_val_metric = val_metric
+
+                    best_val_epoch = e
+                    os.makedirs(os.path.dirname(self.path), exist_ok=True)
+                    torch.save(self.model, self.path + self.temp_file_name)
+
+            # If time limit has exceeded or we haven't improved in some number of epochs, stop early.
+            if e - best_val_epoch > epochs_wo_improve:
+                break
+            if time_limit:
+                time_elapsed = time.time() - start_time
+                time_left = time_limit - time_elapsed
+                if time_left <= 0:
+                    logger.log(20, "\tRan out of time, stopping training early.")
+                    break
+
+        if valloader is not None:
+            try:
+                self.model = torch.load(self.path + self.temp_file_name)
+                os.remove(self.path + self.temp_file_name)
+            except:
+                pass
+            logger.log(15, "Best model found in epoch %d" % best_val_epoch)
+
+    def _fit(self, X_train, y_train, X_val=None, y_val=None, X_unlabeled=None, time_limit=None, **kwargs):
+        import torch
+        self.set_default_params(y_train)
+        num_cols = X_train.shape[1]
+        if num_cols > self.params['max_columns']:
+            raise NotImplementedError(
+                f"This dataset has {num_cols} columns and exceeds 'max_columns' == {self.params['max_columns']}.\n"
+                f"If you are confident you will have enough memory, set 'max_columns' higher and try again.")
```

Comment:
"set 'max_columns'"
-> 
"set the 'max_columns' hyperparameter"

Reply:
Changed error msg.